### DISCUSSION TITLE
Analog output

### DISCUSSION THREAD

**Discussion by Telea-Mihai** [pschatzmann/arduino-audio-tools#284] [Q&A] [answered]
Now first, I want to say I'm new to this audio thing, and this library clearly made it easier to stream files to my esp but is there a way to get this input and output it in an amplifier without I2S. In my country there really aren't any available, I only got these simple amplifiers and I managed to get them working and play sounds. And from what I understand if I just analogWrite the output to the pin, that should just give me the value of the wave of the song. The thing is I cant find any way of doing that with this library. Sorry if my description of my problem wasn't accurate but at this moment I'm a bit frustrated (not with this). Any help is welcome, and thank you. If you need info, ask please.

**Comment by pschatzmann** [comment, marked as answer]
The ESP32 has a built in DAC that you can try to use: The related class is AnalogAudioStream. 
But please note that you might expect to hear quite some noise, but it will be better then a PWM output (PWMAudioStream).

I am usually ordering by external DAC modules from China (from Bangood or Aliexpress) and I would be surprised if you could not get those delivered as well to your country.

**Reply by Telea-Mihai** [reply]
Yeah, its not that I cant find them and they cant be delivered, is just that i pay double the price of the board itself in delivery costs, and if there is an alternative, at least for now, its not worth it. Anyway, thanks I will try your solution!

**Reply by Telea-Mihai** [reply]
I have a problem in compiling with the `MP3DeocderHelix.h`, here is the full error message: `Arduino: 1.8.19 (Windows 10), Board: "ESP32 Dev Module, Disabled, Default 4MB with spiffs (1.2MB APP/1.5MB SPIFFS), 240MHz (WiFi/BT), QIO, 80MHz, 4MB (32Mb), 921600, Core 1, Core 1, None"

Multiple libraries were found for "WiFi.h"

In file included from C:\Users\Placeholder\Documents\Arduino\libraries\arduino-audio-tools-main\examples\examples-stream\streams-url_mp3-analog\streams-url_mp3-analog.ino:14:

 Used: C:\Users\PlaceHolder\AppData\Local\Arduino15\packages\esp32\hardware\esp32\2.0.4\libraries\WiFi

C:\Users\PlaceHolder\Documents\Arduino\libraries\arduino-audio-tools-main\src/AudioCodecs/CodecMP3Helix.h:6:10: fatal error: MP3DecoderHelix.h: No such file or directory

 Not used: C:\Program Files (x86)\Arduino\libraries\WiFi

 #include "MP3DecoderHelix.h"

          ^~~~~~~~~~~~~~~~~~~

compilation terminated.

exit status 1

Error compiling for board ESP32 Dev Module.



This report would have more information with
"Show verbose output during compilation"
option enabled in File -> Preferences.`

Im using the `stream-url_mp3-analog` example. I remember trying the analog generator example and this being a problem as well.
Did I miss something when installing?

**Comment by pschatzmann** [comment]
I think you forgot to install the libhelix project: Here is the link to all [codec projects](https://github.com/pschatzmann/arduino-audio-tools/wiki/Encoding-and-Decoding-of-Audio)

**Reply by Telea-Mihai** [reply]
Thank you very much! And sorry, but where do I set my output pins? I tried to set the star pin attribute of the default config to my gpio but it doesnt work, nowhere in the examples it says anything about the output pins. Sorry, but im stupid. From what i've seen the wiki concentrates on external DAC's but haven't seen any documentation on the internal one and how to set it up. But about the encoding, it worked after installing the library!

**Comment by pschatzmann** [comment]
You can't - they are fixed and the same for every esp32. If you set the log level to info the pins should be printed in the log.
Or just google esp32 internal dac pins

**Reply by Telea-Mihai** [reply]
Got it, thank you!
